### PR TITLE
Migrate to a standalone Tonic repository separated from Topaz

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@chinmaygarde @joshuaseaton

File: lib/ui/dart_wrapper.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// found in the LICENSE file.
+
+#ifndef FLUTTER_LIB_UI_DART_WRAPPER_H_
+#define FLUTTER_LIB_UI_DART_WRAPPER_H_
+
+#include "lib/fxl/memory/ref_counted.h"
+#include "third_party/tonic/dart_wrappable.h"
+
+template <typename T>
+class RefCountedDartWrappable : public fxl::RefCountedThreadSafe<T>,
```
Comment (chinmaygarde) [inline lib/ui/dart_wrapper.h]: Can we move to the fml variant while we are at it anyway?

File: lib/ui/dart_wrapper.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+                                public tonic::DartWrappable {
+ public:
+  virtual void RetainDartWrappableReference() const override {
```
Comment (chinmaygarde) [inline lib/ui/dart_wrapper.h]: Please move the override out-of-line.

File: lib/ui/dart_wrapper.h (hunk 1 above)
Comment (jason-simmons) [reply lib/ui/dart_wrapper.h]: I'd like to defer this to a future PR

File: lib/ui/dart_wrapper.h (hunk 2 above)
Comment (chinmaygarde) [reply lib/ui/dart_wrapper.h]: Nevermind. Didn't see that this was templated.

File: lib/ui/dart_wrapper.h (hunk 1 above)
Comment (chinmaygarde) [reply lib/ui/dart_wrapper.h]: sgtm.

File: lib/ui/dart_wrapper.h [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_LIB_UI_DART_WRAPPER_H_
+#define FLUTTER_LIB_UI_DART_WRAPPER_H_
+
+#include "lib/fxl/memory/ref_counted.h"
+#include "third_party/tonic/dart_wrappable.h"
+
```
[the remark this answers is not in the document]
Comment (jason-simmons) [reply lib/ui/dart_wrapper.h]: Moved this to the blink namespace

File: runtime/dart_isolate.cc [hunk 4]
```diff
@@ -169,7 +173,11 @@ bool DartIsolate::Initialize(Dart_Isolate dart_isolate, bool is_root_isolate) {
     if (auto task_runner = GetTaskRunners().GetUITaskRunner()) {
       // Isolates may not have any particular thread affinity. Only initialize
       // the message handler if a task runner is explicitly specified.
```
[the remark this answers is not in the document]
Comment (jason-simmons) [reply runtime/dart_isolate.cc]: done

**Pull request merged**